packages/core/src/xr/XRManager.ts

```diff
@@ -0,0 +1,303 @@
+import { IXRFeatureDescriptor, IXRPlatformFeature } from "@galacean/engine-design";
+import { XRInputManager } from "./input/XRInputManager";
+import { XRFeatureType } from "./feature/XRFeatureType";
+import { XRSessionManager } from "./session/XRSessionManager";
+import { XRSessionState } from "./session/XRSessionState";
+import { XRSessionType } from "./session/XRSessionType";
+import { XRFeatureManager } from "./feature/XRFeatureManager";
+import { IXRDevice } from "./IXRDevice";
+import { Engine } from "../Engine";
+import { Scene } from "../Scene";
+import { Entity } from "../Entity";
+
+type TXRFeatureManager = XRFeatureManager<IXRFeatureDescriptor, IXRPlatformFeature>;
+type TXRFeatureManagerConstructor = new (engine: Engine) => TXRFeatureManager;
+
+/**
+ * XRManager is the entry point of the XR system.
+ */
+export class XRManager {
+  // @internal
+  static _featureManagerMap: TXRFeatureManagerConstructor[] = [];
+
+  /** Hardware adaptation for XR. */
+  xrDevice: IXRDevice;
```

Comment:
This is internal，What is the value of public?

Reply:
Done.